main.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+while getopts a:b:c:d:e:f:g:h flag
+do
+    case "${flag}" in
+        a) publisher=${OPTARG};;
+        b) credentials=${OPTARG};;
+        c) bucket=${OPTARG};;
+        d) entity=${OPTARG};;
+        e) azureAccountName=${OPTARG};;
+        f) azureAccountKey=${OPTARG};;
+        g) awsRoleArn=${OPTARG};;
+        h) awsEndpoint=${OPTARG};;
+    esac
+done
+
+echo "Generating the docs" 
+techdocs-cli generate --no-docker --verbose
+
+publishCmd="techdocs-cli publish \
+        --publisher-type $publisher \
+        --storage-name $bucket \
+        --entity $entity"
+
+if [[ $publisher == "googleGcs" ]]; then
+    GOOGLE_APPLICATION_CREDENTIALS=$credentials
```

Comment:
Unfortunately, `GOOGLE_APPLICATION_CREDENTIALS` is a filepath. Something like this should work:


```suggestion
    GOOGLE_APPLICATION_CREDENTIALS=/tmp/credentials
    echo $credentials | base64 -d > $GOOGLE_APPLICATION_CREDENTIALS
```

Reply:
Hmm OK. I had thought `If the environment variable GOOGLE_APPLICATION_CREDENTIALS is set, ADC uses the service account key or configuration file that the variable points to` meant you could use it as a proxy for the service account key, as well as a file pointer (https://cloud.google.com/docs/authentication/production)

Reply:
OK I now see the service account key isn't a normal string

Reply:
I've been trying to test this on my machine but it's proving to be a bit of a hassle 😞

Reply:
Confirmed that it has to be a filepath: 

```
$ export GOOGLE_APPLICATION_CREDENTIALS="$(cat ~/.config/gcloud/application_default_credentials.json)" 
$ techdocs-cli publish --publisher-type googleGcs --storage-name "gc-prd-backstage-catalog-staging" --entity="default/Component/backstage"
info: Creating Google Storage Bucket publisher for TechDocs
error: Could not retrieve metadata about the GCS bucket gc-prd-backstage-catalog-staging. Make sure the bucket exists. Also make sure that authentication is setup either by explicitly defining techdocs.publisher.googleGcs.credentials in app config or by using environment variables. Refer to https://backstage.io/docs/features/techdocs/using-cloud-storage
error: from GCS client library: The file at {
  "client_id": "<top secret>",
  "client_secret": "<top secret>",
  "refresh_token": "<top secret>",
  "type": "<top secret>"
} does not exist, or it is not a file.
```